### ISSUE TITLE
controller-gen crd generation does not work across modules layouts

### ISSUE THREAD

**Issue by phantomjinx** [kubernetes-sigs/controller-tools#516]
The operator-sdk [documentation](https://sdk.operatorframework.io/docs/building-operators/golang/quickstart) lays out an operator go project with `api` in the root directory. It also creates a Makefile with the rule `make manifests` defined as
```
$(CONTROLLER_GEN) $(CRD_OPTIONS) rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
```
so relying on controller-gen for generation of the operator CRDs.

While migrating an operator [project](https://github.com/apache/camel-k) to using this CRD generation, there exists an issue with apis that have been modularized, ie. api directory with their own mod.go file.

I've created a [repository](https://github.com/phantomjinx/memcached-operator/tree/master) demonstrating the problem using the memcached-operator example.
- An [api](https://github.com/phantomjinx/memcached-operator/tree/master/api/v1) is generated with a version "v1" and given the kind "memcached";
- A 2nd [api](https://github.com/phantomjinx/memcached-operator/tree/master/api/v1alpha1) is generated with a version "v1alpha1" and given the kind "memcachedext";
- The 2nd api then uses a [type](https://github.com/phantomjinx/memcached-operator/blob/master/api/v1alpha1/memcachedext_types.go#L50) from the v1 api hence placing a parent-child dependency on these apis.

For this to compile correctly, the v1 package had to be imported into the v1alpha1 api. This happened to require the api be [modularized](https://github.com/phantomjinx/memcached-operator/blob/master/api/go.mod). Maybe, this is unnecessary for solving the import problem but is required anyway for making available the api as a module in its own right.

Issues:
- `make manifests` fails to generate any CRDs due to the `controller-gen` command;
- If the go.mod is removed then things do get generated and this seems to be a possible workaround. However, a further problem emerges due to v1alpha1 depending on v1 with the v1 CRD containing a duplicated schema, ie. 2 versions of schema v1.

**Comment by phantomjinx** [comment]
Update on duplicated schema
This appears to be intermittent and maybe dependent on there be a copy of the api directory, eg. in the camel-k project the api is located in `pkg/apis` so a workaround is to copy it to /api first. Consequently, every other execution of controller-gen seems to generate duplicate schema in the CRDs.
Cannot currently mirror this reliably in the example project other than to copy /api down to /pkg/apis and then execute controller-gen with `paths=./...`. In the [camel-k project](https://github.com/phantomjinx/camel-k), duplication does occur even when setting `paths=./api/...`.

**Comment by akutz** [comment]
I am currently bit by this as well. Has anyone encountered a solution in the wild?

**Comment by akutz** [comment]
I filed a PR to fix this issue -- https://github.com/kubernetes-sigs/controller-tools/issues/516.

**Comment by akutz** [comment]
This has been fixed with #663.
